torch_geometric/nn/models/neural_fingerprint.py

```diff
@@ -0,0 +1,73 @@
+import torch
+import torch.nn as nn
+
+from torch_geometric.nn import MessagePassing
+from torch_geometric.utils import add_self_loops
+
+
+class NeuralConv(MessagePassing):
+    def __init__(self, num_features, fingerprint_length):
+        super(NeuralConv, self).__init__(aggr="add")
+        self.H = nn.ModuleList()
+        # H[i] would correspond to the hidden weight which would be used
+        # when num_neighbors=i+1
+        for i in range(5):
+            self.H.append(nn.Linear(num_features, num_features))
+
+        self.W = nn.Linear(num_features, fingerprint_length)
+
+    def forward(self, x, edge_index):
+        edge_index = add_self_loops(edge_index)
+
+        return self.propagate(edge_index, size=(x.size(0), x.size(0)), x=x)
+
+    def message(self, x_i, x_j, edge_index, size):
+        num_neighbors = x_j.size(0)
+
+        return {"msg": x_j, "num_neighbors": num_neighbors}
+
+    def update(self, aggr_out):
+        num_neigbors = aggr_out["num_neighbors"] - 2
```

Comment:
Why is this -2, and not -1? We added only 1 self-loop, right?

Reply:
Yes the number of neighbours would be with -1, but since self.H is zero indexed meaning self.H[ i ] would correspond to i+1 neighbours.